### PR TITLE
Tutorial 8: Async Send and Monitor

### PR DESCRIPTION
**Pull request by lzpap**
## Description
A tutorial on how to use the new asynchronous PyOTA API.

### CODE REVIEW

File: docs/tutorials.rst [hunk 1]
```diff
@@ -716,11 +716,180 @@ Now you know how to use the Tangle for data storage while keeping privacy.
...
+
+.. literalinclude:: ../examples/tutorials/08_async_send_monitor.py
+   :lines: 1-17
+   :lineno-start: 1
+
+Notice, that we import the :py:class:`AsyncIota` api class, because we
+would like to use the asynchronous and concurrent features of PyOTA.
+:py:class:`List` from the :py:class:`typing` library is needed for correct
+type annotations, and we also import the `asyncio`_ library. This will come
+handy when we want to schedule and run the coroutines.
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
in handy when we want to schedule and run the coroutines.
```

File: docs/tutorials.rst [hunk 2]
```diff
@@ -716,11 +716,180 @@ Now you know how to use the Tangle for data storage while keeping privacy.
...
+
+On line 6, we instantiate an asynchronous IOTA API. Functionally, it does the
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
On line 6, we instantiate an asynchronous IOTA api. Functionally, it does the
```

(for consistency with the rest of the docs)

File: docs/tutorials.rst [hunk 3]
```diff
@@ -716,11 +716,180 @@ Now you know how to use the Tangle for data storage while keeping privacy.
...
+On line 28, we send the transfers with the help of
+:py:meth:`AsyncIota.send_transfer`. Since this is not a regular method, but a
+coroutine, we have to ``await`` its result. :py:meth:`AsyncIota.send_transfer`
+takes care of building the bundle, doing proof-of-work and sending the
+transactions within the bundle to the network.
+
+Once we sent the transfer, we collect individual transaction hashes from the
+bundle, which we will use for confirmation checking.
+
+On line 39, the so called confirmation checking starts. With the help of
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
On line 39, the so-called confirmation checking starts. With the help of
```

File: docs/tutorials.rst [hunk 4]
```diff
@@ -716,11 +716,180 @@ Now you know how to use the Tangle for data storage while keeping privacy.
...
+:py:meth:`AsyncIota.get_inclusion_states`, we determine if our transactions
+have been confirmed by the network. The ``None`` value for the ``tips``
+parameter in the argument list basically means that check against the latest
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
parameter in the argument list basically means that we check against the latest
```

File: docs/tutorials.rst [hunk 5]
```diff
@@ -716,11 +716,180 @@ Now you know how to use the Tangle for data storage while keeping privacy.
...
+milestone.
```
Comment (todofixthis) [inline docs/tutorials.rst]: ❔ Is there something in the IOTA docs that we can link to for more info about what are milestones (and also what will change when Coordinator goes away)?

File: docs/tutorials.rst [hunk 6]
```diff
@@ -716,11 +716,180 @@ Now you know how to use the Tangle for data storage while keeping privacy.
...
+
+On line 43, we iterate over our original ``sent_tx_hashes`` list of sent
+transaction hashes and ``git_response['states']``, which is a list of ``bool``
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
transaction hashes and ``gis_response['states']``, which is a list of ``bool``
```

File: docs/tutorials.rst [hunk 7]
```diff
@@ -716,11 +716,180 @@ Now you know how to use the Tangle for data storage while keeping privacy.
...
+values, at the same time using the built in `zip`_ method. We also employ
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
values, at the same time using the built-in `zip`_ method. We also employ
```

File: examples/tutorials/08_async_send_monitor.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    print('Sent bundle with transactions: ')
+    print(*sent_tx_hashes, sep='\n')
+
+    # Measure elapsed time
+    elapsed = 0
+
+    print('Checking confirmation...')
+    while len(sent_tx_hashes) > 0:
+        # Determine if transactions are confirmed
+        git_response = await api.get_inclusion_states(sent_tx_hashes, None)
```
Comment (todofixthis) [inline examples/tutorials/08_async_send_monitor.py]:
```suggestion
        gis_response = await api.get_inclusion_states(sent_tx_hashes, None)
```

File: docs/tutorials.rst (hunk 5 above)
Comment (lzpap) [reply docs/tutorials.rst]: yep, added it in the new version.

File: examples/tutorials/08_async_send_monitor.py (hunk 8 above)
Comment (lzpap) [reply examples/tutorials/08_async_send_monitor.py]: good catch :D

**Pull request merged**